src/ripple/peerfinder/impl/Logic.h

```diff
@@ -401,18 +406,15 @@ class Logic
             return Result::full;
         }
 
-        // Set key and cluster right before adding to the map
-        // otherwise we could assert later when erasing the key.
-        counts_.remove (*slot);
+        // Set the key right before adding to the map, otherwise we might
+        // assert later when erasing the key.
         slot->public_key (key);
-        slot->cluster (cluster);
-        counts_.add (*slot);
-
-        // Add the public key to the active set
-        auto const result = keys_.insert (key);
-        // Public key must not already exist
-        assert (result.second);
-        (void) result.second;
+        {
+            auto const result = keys_.insert(key);
+            // Public key must not already exist
+            assert (result.second);
+            (void)result.second;
```

Comment:
What about space after C-style cast?

Reply:
Ideally, we’d avoid C-style casts altogether. Sad that we don’t have `std::assert` which requires the compiler to treat the thing being asserted as used to avoid issuing warnings.

Reply:
Why can't we use `std::assert`?